harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -122,23 +122,67 @@ def _init_device(self) -> None:
             self.device = torch.device("cpu")
         check.is_not_none(self.device)
 
-    def _init_model_and_optimizer(self) -> None:
+    def _init_model_and_optimizers(self) -> None:
+        # backwards compatibility: remove this when removing the old functions.
+        if util.is_overridden(self.trial._build_optimizers, PyTorchTrial) or util.is_overridden(
+            self.trial._build_lr_schedulers, PyTorchTrial
+        ):
+            check.true(util.is_overridden(self.trial._build_optimizers, PyTorchTrial))
+            check.true(util.is_overridden(self.trial._build_lr_schedulers, PyTorchTrial))
```

Comment:
non-blocking: I think we should add checks that the old versions are not overridden also.